src/camera.cpp

```diff
@@ -484,13 +484,12 @@ void Camera::update(LocalPlayer* player, f32 frametime, f32 busytime,
 
 void Camera::updateViewingRange()
 {
+	f32 viewing_range = g_settings->getFloat("viewing_range");
+	m_draw_control.wanted_range = viewing_range;
 	if (m_draw_control.range_all) {
 		m_cameranode->setFarValue(100000.0);
 		return;
 	}
-
```

Comment:
'R' will still show all _loaded_ blocks. Here I'm just allowing changing the view-range while range_all is active. There is no functional change (before pressing + or - while range_all was active had no effect - even though it would tell you on the screen).

I see your point on 'F3', though. I think what you are saying is that with range_all we should always request all blocks from the server. Is that what you are saying, i.e. with range_all _always retrieve max_block_send_distance range of blocks from the server?

That change would have to be made in `class Client`.

I do find it a but too easy then to pull the maximum amount of data from the server. As it is now you'd have to select range_all _and_ increase the view_range to get up to max_block_send_distance worth of blocks.

Don't feel strongly one way or the other. Tell me what you prefer and I'll do that. :)